pkg_resources/__init__.py

```diff
@@ -691,7 +702,9 @@ def find(self, req: "Requirement"):
             raise VersionConflict(dist, req)
         return dist
 
-    def iter_entry_points(self, group: str, name: Optional[str] = None):
+    def iter_entry_points(
+        self, group: str, name: Optional[str] = None
+    ) -> Generator["EntryPoint", None, None]:
```

Comment:
Should we do `Iterator["EntryPoint"]` here?

I find the `Generator[..., None, None]` annotation very distracting (personal opnion).
Also the name of the method is `iter_*`...

In general I find it conceptually easier to deal with `Iterator[...]`.
It is immediately obvious what to do with the value.

`Generator[...]` on the other hand needs you to remember which position regards the return type, which position regards the yield type and which position regards the send type.

(Again this is a personal opinion)

Reply:
For `Generator[type, None None]` sure.
There's precedence even in typeshed and enforced in Flake8-PYI to consider `__iter__` methods that return `Generators` as an implementation detail and to use `Iterator` instead.

Would you like me to change all of the explicit generator return types that have that signature?

Reply:
(for the sake of this PR I'll split it off, but there's almost nothing left to do to mark `pkg_resources` as py.typed, so that's definitely something I'd like to handle as soon as all PRs linked in this PR's description are merged)